aquadrone_sensors/nodes/battery_sensor.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+import rospy
+import sys
+import smbus
+from random import random
+from sensor_msgs.msg import BatteryState
+from MCP342x import MCP342x
+
+
+RESOLUTION = 18
+R1 = [10.0e3, 47.0e3, 86.6e3, 127.0e3]
+R2 = 28.7e3
+
+
+class MCP342x_Test:
+    def __init__(self, addr, channel, resolution, scale_factor):
+        self.ident = (addr, channel, resolution)
+        self.scale = scale_factor
+
+    def configure(self):
+        print(str(rospy.get_time()) + " " + str(self.ident) + " MCP3424-SIM CONFIGURED")
+        
+    def convert_and_read(self):
+        voltage = 14.8 / 4 * (self.ident[1] + 1) * 0.75 + (random() - 0.5) * 0.4
+        print(str(rospy.get_time()) + " " + str(self.ident) + " " + "{:.2f}".format(voltage) + " MCP3424-SIM CONVERT_AND_READ")
+        return voltage
+
+
+class BatteryPack:
+    def __init__(self, bus, addr, topic):
+        self.pub = rospy.Publisher(topic, BatteryState, queue_size=10)
+        self.volts_raw = [0.0] * 8
+        self.volt_sensors = []
+        for j in range(4):
+            if bus is None:
+                adr_sensor = MCP342x_Test(addr, j, RESOLUTION, R2 / (R1[j]+R2) )
+            else:
+                adr_sensor = MCP342x(bus, addr, j, resolution=RESOLUTION, scale_factor= R2 / (R1[j]+R2) )
+            self.volt_sensors.append(adr_sensor)
+        
+        for s in self.volt_sensors:
+            s.configure()
+        
+    def read(self):
+        self.volts_raw = [s.convert_and_read() for s in self.volt_sensors]
```

Comment:
TODO: also store calculated battery capacity:
there is a way to estimate % charged based on voltage (simple way is to do it linearly. Say 3.2V is 0% and 4V is 100%, average this out for all cells in the pack)